src/ucp/core/ucp_ep.c

```diff
@@ -249,10 +253,14 @@ static UCS_F_NOINLINE void ucp_ep_remove_progress_callbacks(ucp_ep_h ep)
 
 void ucp_ep_destroy_base(ucp_ep_h ep)
 {
-    if (--ep->ref_cnt != 0) {
+    ucp_ep_cnt_dec(&ep->ref_cnt);
```

Comment:
rename ucp_ep_destroy_base to ucp_ep_remove_ref() to -- and destroy if 0
ucp_ep_add_ref() to only ++
if need to destroy unconditionally - use ucp_ep_destroy_base()

add_ref() {
 ++
}

remove_ref {
  if (--ref == 0) destroy
}